webodf/lib/core/DomUtils.js

```diff
@@ -581,15 +581,21 @@
          * Removes all unwanted nodes from targetNode includes itself.
          * @param {!Node} targetNode
          * @param {function(!Node):!boolean} shouldRemove check whether a node should be removed or not
+         * @param {function(!Node):!boolean=} shouldSkip check whether a node and its subtree should be skipped
          * @return {?Node} parent of targetNode
          */
-        function removeUnwantedNodes(targetNode, shouldRemove) {
+        function removeUnwantedNodes(targetNode, shouldRemove, shouldSkip) {
```

Comment:
The introduction of the ``shouldSkip`` makes this look more and more like it should follow the existing TreeWalker pattern of using return values rather than boolean filters (e.g., return NodeFilter.REJECT = delete, NodeFilter.SKIP = skip, NodeFilter.ACCEPT = keep).

The point of ambiguity with the interface here is what happens if ``shouldSkip`` & ``shouldDelete`` are both true?

You could easily fix that in the API documentation, however.